src/main/resources/avro/referencemethods.avdl

```diff
@@ -113,9 +113,9 @@ as JSON.
 */
 record SearchReferencesRequest {
   /**
-  If present, return only references which belong to this reference set.
+  The ReferenceSet to search.
   */
-  union { null, string } referenceSetId = null;
```

Comment:
We need to allow the same reference to be in many reference sets.  For example the same human mitochondrial sequence will be in 
many different versions of the human genome reference set.  Even the same GRCh38 primary chromosome 1 will be in different reference
sets, with and without EBV, alts, IMGD MHC versions etc.  This is essential for practical use.  This has nothing to do with the graph world, it
is core to the way that people practically use genome reference sequences.

Slightly more radically, I think in the graph world we need the  ability to create extended reference sets that include one previous reference
set and add variants to it.  This amounts to a single inheritance tree.  In the side graph level hierarchy I would require the included reference 
to be first, so that other stuff adds to it.  I believe this cleanly captures how we want to build and extend graphs.

Richard

> On 3 Apr 2015, at 09:39, Jerome Kelleher <notifications@github.com> wrote:
> 
> In src/main/resources/avro/referencemethods.avdl <https://github.com/ga4gh/schemas/pull/253#discussion_r27720795>:
> 
> >    */
> > -  union { null, string } referenceSetId = null;
> OK, fair enough. @adamnovak <https://github.com/adamnovak> or @richarddurbin <https://github.com/richarddurbin> could you clarify the status of the many-to-many relationships between references and reference sets in the graph world please?
> 
> (My opinion here would be to try and keep things simple and avoid many-to-many relationships if we can. Implementations are free to reuse the underlying objects representing the same references within their code, but can present them to the outside world as-if they were different objects.)
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ga4gh/schemas/pull/253/files#r27720795>.
> 




-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.